core/sr-io/src/lib.rs

```diff
@@ -46,352 +65,774 @@ pub enum EcdsaVerifyError {
+		let storage_key = child_storage_key_or_panic(child_storage_key);
+		self.set_child_storage(storage_key, key.to_vec(), value.to_vec());
+	}
+
+	/// Clear the storage of the given `key` and its value.
+	fn clear(&mut self, key: &[u8]) {
+		self.clear_storage(key)
+	}
+
+	/// Clear the given child storage of the given `key` and its value.
+	fn child_clear(&mut self, child_storage_key: &[u8], key: &[u8]) {
+		let storage_key = child_storage_key_or_panic(child_storage_key);
+		self.clear_child_storage(storage_key, key);
+	}
+
+	/// Clear an entire child storage.
+	fn child_storage_kill(&mut self, child_storage_key: &[u8]) {
+		let storage_key = child_storage_key_or_panic(child_storage_key);
+		self.kill_child_storage(storage_key);
+	}
+
+	/// Check whether the given `key` exists in storage.
+	fn exists(&self, key: &[u8]) -> bool {
+		self.exists_storage(key)
+	}
+
+	/// Check whether the given `key` exists in storage.
+	fn child_exists(&self, child_storage_key: &[u8], key: &[u8]) -> bool {
+		let storage_key = child_storage_key_or_panic(child_storage_key);
+		self.exists_child_storage(storage_key, key)
+	}
+
+	/// Clear the storage of each key-value pair where the key starts with the given `prefix`.
+	fn clear_prefix(&mut self, prefix: &[u8]) {
+		Externalities::clear_prefix(*self, prefix)
+	}
+
+	/// Clear the child storage of each key-value pair where the key starts with the given `prefix`.
+	fn child_clear_prefix(&mut self, child_storage_key: &[u8], prefix: &[u8]) {
+		let storage_key = child_storage_key_or_panic(child_storage_key);
+		self.clear_child_prefix(storage_key, prefix);
+	}
+
+	/// "Commit" all existing operations and compute the resulting storage root.
+	fn root(&mut self) -> H256 {
+		self.storage_root()
+	}
+
+	/// "Commit" all existing operations and compute the resulting child storage root.
+	fn child_root(&mut self, child_storage_key: &[u8]) -> Vec<u8> {
+		let storage_key = child_storage_key_or_panic(child_storage_key);
+		self.child_storage_root(storage_key)
+	}
+
+	/// "Commit" all existing operations and get the resulting storage change root.
+	fn changes_root(&mut self, parent_hash: [u8; 32]) -> Option<H256> {
+		self.storage_changes_root(parent_hash.into()).ok().and_then(|h| h)
+	}
+
+	/// A trie root formed from the iterated items.
+	fn blake2_256_trie_root(input: Vec<(Vec<u8>, Vec<u8>)>) -> H256 {
+		Layout::<primitives::Blake2Hasher>::trie_root(input)
+	}
+
+	/// A trie root formed from the enumerated items.
+	fn blake2_256_ordered_trie_root(input: Vec<Vec<u8>>) -> H256 {
+		Layout::<primitives::Blake2Hasher>::ordered_trie_root(input)
+	}
+}
+
+/// Interface that provides miscellaneous functions for communicating between the runtime and the node.
+#[runtime_interface]
+pub trait Misc {
+	/// The current relay chain identifier.
+	fn chain_id(&self) -> u64 {
+		externalities::Externalities::chain_id(*self)
+	}
+
+	/// Print a number.
+	fn print_num(val: u64) {
+		log::debug!(target: "runtime", "{}", val);
```

Comment:
Shouldn't this be an `info`? I think now by default it's not going to display anything, and previously `print` was a major way to understand what's happening in the runtime.

Reply:
This was done actually here: https://github.com/paritytech/substrate/commit/0eeee4ddeeebf6e1a2edbb987113eb9847db7040

I think that if you need to debug the runtime `RUST_LOG=runtime=debug` is probably better than people always asking what kind of output this is.